addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairIntelliFloHandler.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.pentair.handler;
+
+import static org.openhab.binding.pentair.PentairBindingConstants.*;
+
+import java.math.BigDecimal;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.pentair.PentairBindingConstants;
+import org.openhab.binding.pentair.internal.PentairPacket;
+import org.openhab.binding.pentair.internal.PentairPacketPumpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PentairIntelliFloHandler} is responsible for implementation of the Intelliflo Pump. This will
+ * parse/dispose of
+ * status packets to set the stat for various channels.
+ *
+ * @author Jeff James - Initial contribution
+ */
+public class PentairIntelliFloHandler extends PentairBaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(PentairIntelliFloHandler.class);
+    protected PentairPacketPumpStatus ppscur = new PentairPacketPumpStatus();
+
+    public PentairIntelliFloHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Intelliflo - Thing ID: {}.", this.getThing().getUID());
+
+        id = ((BigDecimal) getConfig().get("id")).intValue();
+
+        updateStatus(ThingStatus.ONLINE);
+
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Thing {} disposed.", getThing().getUID());
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
please do not set the status anymore in the dispose - the framework takes care of that.